scripts/ci/stats/merged_prs.py

```diff
@@ -84,56 +74,72 @@ def main():
         prj['commits'] = pr.commits
         # The branch we are targeting. main vs release branches.
         prj['base'] = pr.base.ref
+        # defaults
+        prj['review_rule'] = "no"
+        prj['time_rule'] = False
 
-        ll = []
-        for l in pr.labels:
-            ll.append(l.name)
-        prj['labels'] = ll
+        labels = []
+        for label in pr.labels:
+            labels.append(label.name)
+        prj['labels'] = labels
 
-        # take first assignee, otherwise we have no assignees and this rule is not applicable
-        if pr.assignee:
-            prj['assignee'] = pr.assignee.login
-        else:
-            prj['assignee'] = "none"
-            prj['reviewed_by_assignee'] = "na"
-            prj['review_rule'] = "na"
+        reviewers = set()
+        for review in reviews:
+            # get list of all approved reviews
+            if review.user and review.state == 'APPROVED':
+                reviewers.add(review.user.login)
 
-        # go through all assignees and check if anyone has approved and reset assignee to the one who approved
         for assignee in pr.assignees:
             if assignee.login in reviewers:
+                assignee_reviews += 1
+                # was reviewed at least by one assignee
                 prj['assignee'] = assignee.login
-            elif assignee.login == pr.user.login:
-                prj['reviewed_by_assignee'] = "yes"
+                prj['review_rule'] = "yes"
+
+            # if assignee is the submitter, rule does not apply.
+            if  assignee.login == pr.user.login:
+                prj['review_rule'] = "na"
```

Comment:
I think this could result in "na" or "yes" depending on the order of assignee, maybe just set `review_rule` once outside of the loop depending on the value of `assignee_reviews`? It's also hard to follow right now sicne this can be overridden by a bunch of different code paths

Reply:
updated